src/app/features/location/location.component.ts

```diff
@@ -68,4 +72,39 @@ export class LocationComponent {
   navigate(page: PageEvent) {
     this.navigationService.navigate(page.pageIndex);
   }
+
+  private findNewestMeetup$() {
+    return combineLatest([
+      this.currentPage$,
+      this.location$,
+    ]).pipe(
+      switchMap(([_, loc]) => 
+        this.postsService.getPosts(
+          undefined,
+          undefined,
+          false,
+          post => this.isMeetupCategoryByLocation(post, loc),
+          (a, b) => this.compareByDate(a, b)
+        ).pipe(
+          map(result => {
+            const startOfToday = new Date();
+            startOfToday.setHours(0, 0, 0, 0);
+
+            return this.postsService.findSoonestPostAfter(result.posts, startOfToday.getTime());
+          })
+        )
+      )
+    );
+  }
+
+  private isMeetupCategoryByLocation(post: Post, location: string | null): boolean {
+    return (post.category as string) === 'meetups' && post.location === location;
```

Comment:
Please avoid unnecessary type casting

```suggestion
    return post.category === Category.meetups && post.location === location;
```